.circleci/config.yml

```diff
@@ -38,177 +38,232 @@ defaults: &defaults
-
-            - store_artifacts:
-                path: /tmp/artifacts
-                destination: artifacts
+  configure_env_vars:
+    <<: *defaults
+
+    steps:
+      - checkout
+
+      - run:
+          # Set TERMINUS_ENV and related environment variables.
+          # https://github.com/pantheon-systems/docker-build-tools-ci/blob/6.x/scripts/set-environment
+          name: setup-environment-vars
+          command: /build-tools-ci/scripts/set-environment
+
+      - run:
+          name: clone-bash-env
+          command: cp $BASH_ENV bash_env.txt
+
+      # https://circleci.com/docs/2.0/workflows/#using-workspaces-to-share-data-among-jobs
+      - persist_to_workspace:
+          root: .
+          paths:
+            - bash_env.txt
+
+  static_tests:
+    <<: *defaults
+
+    steps:
+      - checkout
+
+      - restore_cache:
+          keys:
+            - composer-cache-{{ checksum "composer.lock" }}
+            - composer-cache-
+
+      - run:
+          # Set TERMINUS_ENV and related environment variables.
+          # https://github.com/pantheon-systems/docker-build-tools-ci/blob/6.x/scripts/set-environment
+          name: setup-environment-vars
+          command: /build-tools-ci/scripts/set-environment
+
+      - run:
+          name: run static tests
+          command: ./.ci/test/static/run
+
+      - save_cache:
+          key: composer-cache-{{ checksum "composer.lock" }}
+          paths:
+            - $HOME/.composer/cache
+
+  build_php:
+    <<: *defaults
+    steps:
+      - checkout
+
+      - restore_cache:
+          keys:
+            - composer-cache-{{ checksum "composer.lock" }}
+            - composer-cache-
+            - vendor-{{ checksum "composer.lock" }}
+            - vendor-
+
+      - run:
+          # Set TERMINUS_ENV and related environment variables.
+          # https://github.com/pantheon-systems/docker-build-tools-ci/blob/6.x/scripts/set-environment
+          name: setup-environment-vars
+          command: /build-tools-ci/scripts/set-environment
+
+      - run:
+          name: run PHP build step
+          command: ./.ci/build/php
+
+      - save_cache:
+          key: vendor-{{ checksum "composer.lock" }}
+          paths:
+            - $CIRCLE_WORKING_DIRECTORY/vendor
+
+      - persist_to_workspace:
+          root: .
+          paths:
+            - web
+            - vendor
+            - drush
+
+  deploy_to_pantheon:
+    <<: *defaults
+    steps:
+      - checkout
+
+      # Attach the workspace
+      - attach_workspace:
+          at: /tmp
+
+      - run:
+          # Set TERMINUS_ENV and related environment variables.
+          # https://github.com/pantheon-systems/docker-build-tools-ci/blob/6.x/scripts/set-environment
+          name: setup-environment-vars
+          command: /build-tools-ci/scripts/set-environment
+
+      # Rsync the built web directory
+      - run:
+          name: sync web
+          command: rsync -az /tmp/web/ web
```

Comment:
Should be able to do this all in one rsync. Presuming that there is nothing in the workspace except what you want to restore, then `rsync -az /tmp/ .` should do the trick. I guess you have persisted the bash env; you could simply ignore that with an `--exclude`.

Reply:
That's what I was asking with

> Do you think we should persist and rsync the entire docroot instead of specific directories?

Reply:
@greg-1-anderson after reading [the CircleCI doc on `persist_to_workspace`](https://circleci.com/docs/2.0/configuration-reference/#persist_to_workspace) and seeing

> If multiple concurrent jobs persist the same filename then attaching the workspace will error.

I don't think we should persist the whole docroot. While it will be fine for a single build job, if multiple build jobs persist the entire docroot the build will fail as they will persist files with the same name.

The way it is set up currently, each build job only persists the files it builds. While this is a bit more work to maintain the list of files to persist for each build job it is the only way to avoid a conflict.